### PR TITLE
Add Status-Page Subscription Button

### PR DESCRIPTION
**Pull request by mabezi**
Related to #458

### CODE REVIEW

**Comment by mabezi** [comment]
It seems like PyCharm reformatted some line breaks...

File: sipa/translations/en/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -1086,6 +1086,15 @@ msgstr "Incoming"
...
 
+msgid "Status-Page abonnieren"
+msgstr "Subscribe to Status-Page"
+
+msgid "Du wurdest zur Status-Page hinzugefügt. Du bekommst eine E-Mail mit weiteren Details."
+msgstr "You have been added to the status page. You will receive an email with more details."
+
+msgid "Du hast die Statuspage bereits abonniert."
+msgstr "You have already subscribed to the status page."
+
```
Comment (lukasjuhrich) [inline sipa/translations/en/LC_MESSAGES/messages.po:1097]: :+1: for figuring out and adding translations!

File: sipa/utils/__init__.py [hunk 2]
```diff
@@ -159,6 +159,25 @@ def meetingcal():
     return next_meetings
 
 
+def subscribeStatusPage(url: str, email: str) -> bool | None:
```
Comment (lukasjuhrich) [inline sipa/utils/__init__.py:162]: Please use snail-case function names according to [PEP8](https://peps.python.org/pep-0008/#function-and-variable-names):
```suggestion
def subscribe_to_status_page(url: str, email: str) -> bool | None:
```

File: sipa/utils/__init__.py [hunk 3]
```diff
@@ -159,6 +159,25 @@ def meetingcal():
...
+    try:
+        response = requests.post(url, timeout=1,
+                                 headers={
+                                     "Authorization": "Token " + current_app.config['STATUS_PAGE_AUTH'],
+                                     "Content-Type": "application/json; charset=utf8"
+                                 },
+                                 json={"email": email})
```
Comment (lukasjuhrich) [inline sipa/utils/__init__.py:169]: Ideally, use `darker` or something else to properly format this. Not breaking after an opening parenthesis causes alignment quite far to the right, so just line-breaking and indenting once is the better choice here.
```suggestion
        response = requests.post(
            url,
            timeout=1,
            headers={
                "Authorization": "Token " + current_app.config['STATUS_PAGE_AUTH'],
                "Content-Type": "application/json; charset=utf8"
            },
            json={"email": email}
        )
```
Also, check whether the content-type is necessary; I'd guess that if you pass `json` instead of `data` this should be added automatically.

File: sipa/utils/__init__.py [hunk 4]
```diff
@@ -159,6 +159,25 @@ def meetingcal():
     return next_meetings
 
 
+def subscribeStatusPage(url: str, email: str) -> bool | None:
+    try:
+        response = requests.post(url, timeout=1,
+                                 headers={
+                                     "Authorization": "Token " + current_app.config['STATUS_PAGE_AUTH'],
```
Comment (lukasjuhrich) [inline sipa/utils/__init__.py:166]: Please be consistent and use a parameter for this like in the other cases instead of accessing `current_app`

File: sipa/utils/__init__.py [hunk 5]
```diff
@@ -159,6 +159,25 @@ def meetingcal():
     return next_meetings
 
 
+def subscribeStatusPage(url: str, email: str) -> bool | None:
+    try:
+        response = requests.post(url, timeout=1,
```
Comment (lukasjuhrich) [inline sipa/utils/__init__.py:164]: It might be a good idea to make the timeout configurable as well if you finish the feature

File: sipa/utils/__init__.py [hunk 6]
```diff
@@ -159,6 +159,25 @@ def meetingcal():
...
+                                 json={"email": email})
+    except requests.exceptions.RequestException:
+        logger.exception("Error when sending request to %s", url)
+        return None
+    if response.status_code == 400:
+        return False
+    if response.status_code == 201:
+        return True
+
+    return None
```
Comment (lukasjuhrich) [inline sipa/utils/__init__.py:178]: Can you add some exception reporting here as well? a la „unexpected status code $blah“.

File: sipa/utils/__init__.py [hunk 7]
```diff
@@ -159,6 +159,25 @@ def meetingcal():
...
+def subscribeStatusPage(url: str, email: str) -> bool | None:
+    try:
+        response = requests.post(url, timeout=1,
+                                 headers={
+                                     "Authorization": "Token " + current_app.config['STATUS_PAGE_AUTH'],
+                                     "Content-Type": "application/json; charset=utf8"
+                                 },
+                                 json={"email": email})
+    except requests.exceptions.RequestException:
+        logger.exception("Error when sending request to %s", url)
```
Comment (lukasjuhrich) [inline sipa/utils/__init__.py:171]: :+1: for proper log formatting

File: sipa/config/default.py [hunk 8]
```diff
@@ -114,3 +114,7 @@
 
 # link to clandar
 MEETINGS_ICAL_URL = "https://agdsn.de/cloud/remote.php/dav/public-calendars/bgiQmBstmfzRdMeH?export"
+
+# statuspage
+STATUS_PAGE_ENDPOINT = "https://status.agdsn.net/api/subscribers/subscribers/"
+STATUS_PAGE_AUTH = ""
```
Comment (lukasjuhrich) [inline sipa/config/default.py:120]: trailing newline missing

File: sipa/blueprints/usersuite.py [hunk 9]
```diff
@@ -148,15 +148,32 @@ def contact():
...
         gettext("Zahlungsempfänger"): details.recipient,
         gettext("Bank"): details.bank,
         gettext("IBAN"): details.iban,
         gettext("BIC"): details.bic,
         gettext("Verwendungszweck"): details.purpose,
-        gettext("Betrag"): format_currency(months * current_app.config['MEMBERSHIP_CONTRIBUTION'] / 100, 'EUR',
-                                           locale='de_DE')
+        gettext("Betrag"): format_currency(
+            months * current_app.config['MEMBERSHIP_CONTRIBUTION'] / 100, 'EUR',
+            locale='de_DE')
```
Comment (lukasjuhrich) [inline sipa/blueprints/usersuite.py:176]: Starting here you have lots of unrelated changes. please revert these, this makes `git blame` and PR review a confusing.
If you want to improve formatting that's alright, but better done just on `develop` but in any case in a separate commit. :)

File: sipa/config/default.py (hunk 8 above)
Comment (mabezi) [reply sipa/config/default.py]: PyCharm somehow swallows the line break at the end of the file
